src/app.cpp

```diff
@@ -156,9 +156,21 @@ PYBIND11_MODULE(py_dismech, m) {
         .def("applyInitialVelocities", &softRobots::applyInitialVelocities, py::arg("limb_idx"), py::arg("velocities"))
         .def("setup", &softRobots::setup)
         .def("addController", &softRobots::addController, py::arg("controller"))
-        .def_readwrite("limbs", &softRobots::limbs)
-        .def_readwrite("joints", &softRobots::joints)
-        .def_readwrite("controllers", &softRobots::controllers);
+        .def_readonly("limbs", &softRobots::limbs, py::return_value_policy::reference_internal)
+        .def_readonly("joints", &softRobots::joints,
+                      py::return_value_policy::reference_internal)
+        .def_readonly("controllers", &softRobots::controllers);
```

Comment:
I think now the CPP codes do not need to have controllers.

Reply:
Removed.